torch/testing/_internal/common_modules.py

```diff
@@ -175,13 +176,14 @@ def module_inputs_torch_nn_Linear(module_info, device, dtype, requires_grad, **k
 
 def module_inputs_torch_nn_NLLLoss(module_info, device, dtype, requires_grad, **kwargs):
     make_input = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    make_weight = partial(make_tensor, device=device, dtype=dtype, requires_grad=False)
 
     cases: List[Tuple[str, dict]] = [
         ('', {}),
         ('ignore_index', {'ignore_index': 2}),
-        ('weights', {'weight': make_input(10)}),
-        ('weights_ignore_index', {'weight': make_input(10), 'ignore_index': 2}),
-        ('weights_ignore_index_neg', {'weight': make_input(10), 'ignore_index': -1})
+        ('weights', {'weight': make_weight(10).abs()}),
+        ('weights_ignore_index', {'weight': make_weight(10).abs(), 'ignore_index': 2}),
+        ('weights_ignore_index_neg', {'weight': make_weight(10).abs(), 'ignore_index': -1})
```

Comment:
Are these `abs()` required due to #64256 ?

Reply:
Yup. Although... is there a use case for negative weights? 🤔

Reply:
Don't we prefer to have at least one test with negative weights and an explicit skip for the gradgrad for it?
That way we make sure that the gradcheck works and it is clear that the gradgrad is problematic.

Reply:
Yeah, I wonder if we want to revisit the idea of separating metadata for samples that can be used at test collection time from the sample values so we can skip and run things on a per-sample basis. It'd be awesome to have a separate named test for each sample.

For this scenario, if we had separate tests for gradcheck and gradgradcheck, we could then skip either of them for a given sample (although it's probably not likely we'd want to skip gradcheck and run gradgradcheck).

Reply:
The thing that we can use also is just split the ModuleInfo in two: the one with negative weights skips gradgrad and the one with positive weights only runs it.